node_modules/col-assets/lib/api.js

```diff
@@ -267,6 +369,246 @@ var createLink = module.exports.createLink = function(ctx, title, url, opts, cal
   });
 };
 
+/* FILE UPLOAD */
+
+/**
+ * Upload a file to a course in Canvas
+ *
+ * @param  {Context}        ctx                   Standard context containing the current user and the current course
+ * @param  {String}         filePath              The path of the file that should be uploaded
+ * @param  {Function}       callback              Standard callback function
+ * @param  {Object}         callback.err          An error that occurred, if any
+ * @api private
+ */
+var uploadFileToCanvas = function(ctx, filePath, callback) {
+  // Step 1: Tell Canvas that we want to upload a file. Canvas will give us some information about
+  // where we should ship the file to in step 2
+  notifyCanvasOfUpload(ctx, filePath, function(err, uploadInfo) {
+    if (err) {
+      return callback(err);
+    }
+
+    // Step 2: Upload the file to wherever Canvas wants us to upload it. In development this will
+    // be your local canvas instance, in production this will probably be an Amazon S3 bucket
+    uploadFile(ctx, filePath, uploadInfo, function(err, confirmUrl) {
+      if (err) {
+        return callback(err);
+      }
+
+      // Step 3: Confirm the file was uploaded to Canvas
+      confirmFile(ctx, confirmUrl, function(err, file) {
+        if (err) {
+          return callback(err);
+        }
+
+        // Step 4: Hide the file so it doesn't show up in the Files Tool for students
+        return hideFile(ctx, file, callback);
+      });
+    });
+  });
+};
+
+/**
+ * Hide a file. This will ensure that students can't see the uploaded file in the Files tool.
+ *
+ * @param  {Context}        ctx                   Standard context containing the current user and the current course
+ * @param  {Object}         file                  The file information for the uploaded file as described by the Canvas API
+ * @param  {Function}       callback              Standard callback function
+ * @param  {Object}         callback.err          An error that occurred, if any
+ * @api private
+ */
+var hideFile = function(ctx, file, callback) {
```

Comment:
Can you put these functions in the order that they're called in `uploadFileToCanvas`?